src/py/flwr/common/message_test.py

```diff
@@ -202,3 +202,36 @@ def test_repr(cls: type, kwargs: dict[str, Any]) -> None:
 
     # Assert
     assert str(actual) == str(expected)
+
+
+@pytest.mark.parametrize(
+    "message_creation_fn,initial_ttl,reply_ttl,expected_max_ttl",
+    [
+        # Case where the reply_ttl is larger than the allowed TTL
+        (create_message_with_content, 20, 30, 20),
+        (create_message_with_error, 20, 30, 20),
+        # Case where the reply_ttl is within the allowed range
+        (create_message_with_content, 20, 10, 10),
+        (create_message_with_error, 20, 10, 10),
+    ],
+)
+def test_reply_ttl_limitation(
+    message_creation_fn: Callable[[float], Message],
+    initial_ttl: float,
+    reply_ttl: float,
+    expected_max_ttl: float,
+) -> None:
+    """Test that the reply TTL does not exceed the allowed TTL."""
+    message = message_creation_fn(initial_ttl)
+
+    time.sleep(0.1)
+
+    if message.has_error():
+        dummy_error = Error(code=0, reason="test error")
+        reply_message = message.create_error_reply(dummy_error, ttl=reply_ttl)
+    else:
+        reply_message = message.create_reply(content=RecordSet(), ttl=reply_ttl)
+
+    assert (
+        reply_message.metadata.ttl <= expected_max_ttl
```

Comment:
I think we should only limit the TTL if it exceeds the allowed maximum. We should also prevent our code make the actual TTL unnecessarily small.
```suggestion
        abs(reply_message.metadata.ttl - expected_reply_ttl) <= 1e-9
```